style/js/shield.js

```diff
@@ -245,8 +247,11 @@ export function missingIconLoader(map, e) {
     ctx = colorCtx;
   }
 
-  var imgData = ctx.getImageData(0, 0, ctx.canvas.width, ctx.canvas.height);
+  uploadShieldToSpriteSheet(map, id, ctx);
```

Comment:
Let’s refactor this method to avoid early returns. Declare `imgData` at the top. If the conditions aren’t met, generate the blank image here. We’d end up calling `getImageData()` and `addImage()` no matter what.

Reply:
Thanks!  This refactor is a significant improvement in readability.